miden-tx/src/executor/mod.rs

```diff
@@ -34,22 +35,24 @@ pub use data_store::DataStore;
 /// The [TransactionExecutor::execute_transaction()] method is the main entry point for the
 /// executor and produces an [ExecutedTransaction] for the transaction. The executed transaction
 /// can then be used to by the prover to generate a proof transaction execution.
-pub struct TransactionExecutor<D> {
+pub struct TransactionExecutor<D, A> {
     data_store: D,
     compiler: TransactionCompiler,
     exec_options: ExecutionOptions,
+    authenticator: A,
 }
```

Comment:
minor nit: I would probably move the `authenticator` field up to be right after `data_store`.

Also, I'm wondering if we should make it optional - i.e., have transaction executor look like this:
```Rust
pub struct TransactionExecutor<D, A> {
    data_store: D,
    authenticator: Option<A>,
    compiler: TransactionCompiler,
    exec_options: ExecutionOptions,
}
```
One potential benefit is that we won't need to implement `TransactionAuthenticator` for `()` and this will make the intents more explicit. But I'm not sure how much this complicates things elsewhere.

Reply:
I did this, but Rust will have to know the type even when instantiating with `None`, so it still makes sense to implement the trait for `()` IMO (if we were to use the `BasicAuthenticator` then you need an `R:Rng`)